code/game/machinery/telecomms/computers/message.dm

```diff
@@ -80,392 +78,205 @@
+	linkedServer = null
 	return ..()
 
-/obj/machinery/computer/message_monitor/ui_interact(mob/living/user)
-	. = ..()
-	//If the computer is being hacked or is emagged, display the reboot message.
-	if(hacking || (obj_flags & EMAGGED))
-		message = rebootmsg
-	var/dat = "<center><font color='blue'[message]</font></center>"
-
-	if(auth)
-		dat += "<h4><dd><A href='?src=[REF(src)];auth=1'>&#09;<font color='green'>\[Authenticated\]</font></a>&#09;/"
-		dat += " Server Power: <A href='?src=[REF(src)];active=1'>[linkedServer?.on ? "<font color='green'>\[On\]</font>":"<font color='red'>\[Off\]</font>"]</a></h4>"
-	else
-		dat += "<h4><dd><A href='?src=[REF(src)];auth=1'>&#09;<font color='red'>\[Unauthenticated\]</font></a>&#09;/"
-		dat += " Server Power: <u>[linkedServer?.on ? "<font color='green'>\[On\]</font>":"<font color='red'>\[Off\]</font>"]</u></h4>"
-
-	if(hacking || (obj_flags & EMAGGED))
-		screen = MSG_MON_SCREEN_HACKED
-	else if(!auth || LINKED_SERVER_NONRESPONSIVE)
-		if(LINKED_SERVER_NONRESPONSIVE)
-			message = noserver
-		screen = MSG_MON_SCREEN_MAIN
+/obj/machinery/computer/message_monitor/ui_data(mob/user)
+	var/list/data = list(
+		"screen" = screen,
+		"error_message" = error_message,
+		"notice_message" = notice_message,
+		"success_message" = success_message,
+		"auth" = auth,
+		"server_status" = !LINKED_SERVER_NONRESPONSIVE
+	)
 
 	switch(screen)
-		//Main menu
 		if(MSG_MON_SCREEN_MAIN)
-			//&#09; = TAB
-			var/i = 0
-			dat += "<dd><A href='?src=[REF(src)];find=1'>&#09;[++i]. Link To A Server</a></dd>"
-			if(auth)
-				if(LINKED_SERVER_NONRESPONSIVE)
-					dat += "<dd><A>&#09;ERROR: Server not found!</A><br></dd>"
-				else
-					dat += "<dd><A href='?src=[REF(src)];view_logs=1'>&#09;[++i]. View Message Logs </a><br></dd>"
-					dat += "<dd><A href='?src=[REF(src)];view_requests=1'>&#09;[++i]. View Request Console Logs </a></br></dd>"
-					dat += "<dd><A href='?src=[REF(src)];clear_logs=1'>&#09;[++i]. Clear Message Logs</a><br></dd>"
-					dat += "<dd><A href='?src=[REF(src)];clear_requests=1'>&#09;[++i]. Clear Request Console Logs</a><br></dd>"
-					dat += "<dd><A href='?src=[REF(src)];pass=1'>&#09;[++i]. Set Custom Key</a><br></dd>"
-					dat += "<dd><A href='?src=[REF(src)];msg=1'>&#09;[++i]. Send Admin Message</a><br></dd>"
+			data["password"] = password
+			data["status"] = linkedServer.on
+			// Check is AI or cyboeg malf
+			var/mob/living/silicon/silicon_user = user
+			if(istype(silicon_user) && silicon_user.hack_software)
+				data["is_malf"] = TRUE
 			else
-				for(var/n = ++i; n <= optioncount; n++)
-					dat += "<dd><font color='blue'>&#09;[n]. ---------------</font><br></dd>"
-			var/mob/living/silicon/S = usr
-			if(istype(S) && S.hack_software)
-				//Malf/Traitor AIs can bruteforce into the system to gain the Key.
-				dat += "<dd><A href='?src=[REF(src)];hack=1'><i><font color='Red'>*&@#. Bruteforce Key</font></i></font></a><br></dd>"
-			else
-				dat += "<br>"
+				data["is_malf"] = FALSE
```

Comment:
This shoul dall probably become `data["is_malf"] = istype(silicon_user) && silicon_user.hack_software`